sparkmagic/sparkmagic/kernels/kernelmagics.py

```diff
@@ -345,12 +348,13 @@ def _do_not_call_change_endpoint(self, line, cell="", local_ns=None):
             error = u"Cannot change the endpoint if a session has been started."
             raise BadUserDataException(error)
 
-        self.endpoint = Endpoint(server, username, password)
+        self.endpoint = Endpoint(server, authentication, username, password)
 
     def refresh_configuration(self):
         credentials = getattr(conf, 'base64_kernel_' + self.language + '_credentials')()
-        (username, password, url) = (credentials['username'], credentials['password'], credentials['url'])
-        self.endpoint = Endpoint(url, username, password)
+        (username, password, authentication, url) = (credentials['username'], credentials['password'], \
+                credentials['authentication'], credentials['url'])
```

Comment:
Can you please update the README.md file with examples on how to configure different authentication methods in the config file, and list Kerberos support as a feature 😄 ?